src/service/viewer-impl.js

```diff
@@ -972,6 +972,12 @@ export class Viewer {
  * @private
  */
 function parseParams_(str, allParams) {
+  // Temporary workaround for a viewer sending a bad URL
+  // that has two # inside of it.
+  // See Google bug b/33706977
+  // If there is a # (but not at the start of the string)
+  // replace it with & before parsing.
+  str = str.replace(/(.)\#/, '$1&');
```

Comment:
We need to be very careful about whether this string has already been percent decoded. One of the browsers decodes `location.hash`, and I have no idea how they handle `iframe.name`.